### ISSUE TITLE
Multi master cluster can not be reached via kubectl ,  if %50 of masters fails/not accessable

### ISSUE THREAD

**Issue by unlenen** [rancher/rke#2140]
Hi , 

We use k8s (v1.15.11) which is created by rke . We need a HA kubernetes solution  so we created 1 master and n worker at each location. If we lost l %34 of masters , we discovered that we can not access to our cluster with using kubectl . According to "kubectl get cs" , if we reboot %34 of master or stop docker service , other etcd goes to unhealthy status and can not response our kubectl commands. We need thet at least one master available , we should have acccess to cluster. 

Thanks 

Nebi Volkan UNLENEN

ex1 :  6 master 
-------------------------------------------------------------------------------
#kubectl get cs
NAME                 STATUS    MESSAGE             ERROR
scheduler            Healthy   ok
controller-manager   Healthy   ok
etcd-1               Healthy   {"health":"true"}
etcd-2               Healthy   {"health":"true"}
etcd-0               Healthy   {"health":"true"}
etcd-5               Healthy   {"health":"true"}
etcd-3               Healthy   {"health":"true"}
etcd-4               Healthy   {"health":"true"}

if we close 5 master . Result is below . 
#kubectl get cs
NAME                 STATUS      MESSAGE                                                                                             ERROR
etcd-3               Unhealthy   Get https://192.168.21.68:2379/health: dial tcp 192.168.21.68:2379: connect: connection refused
etcd-5               Unhealthy   Get https://192.168.21.110:2379/health: dial tcp 192.168.21.110:2379: connect: connection refused
etcd-1               Unhealthy   Get https://192.168.21.93:2379/health: dial tcp 192.168.21.93:2379: connect: connection refused
etcd-2               Unhealthy   Get https://192.168.21.91:2379/health: dial tcp 192.168.21.91:2379: connect: connection refused
etcd-0               Unhealthy   Get https://192.168.21.109:2379/health: dial tcp 192.168.21.109:2379: connect: connection refused
scheduler            Healthy     ok
controller-manager   Healthy     ok
etcd-4               Unhealthy   HTTP probe failed with statuscode: 503
-------------------------------------------------------------------------------

if we close 3 master . We can not access to cluster again

#kubectl get cs
NAME                 STATUS      MESSAGE                                                                                             ERROR
etcd-1               Unhealthy   Get https://192.168.21.110:2379/health: dial tcp 192.168.21.110:2379: connect: connection refused
etcd-2               Unhealthy   Get https://192.168.21.109:2379/health: dial tcp 192.168.21.109:2379: connect: connection refused
controller-manager   Healthy     ok
etcd-4               Unhealthy   Get https://192.168.21.91:2379/health: dial tcp 192.168.21.91:2379: connect: connection refused
scheduler            Healthy     ok
etcd-0               Unhealthy   HTTP probe failed with statuscode: 503
etcd-5               Unhealthy   HTTP probe failed with statuscode: 503
etcd-3               Unhealthy   HTTP probe failed with statuscode: 503

--------------------------------------------------------------------------------
if we close 2 master  . We can access to our cluster with kubectl commands.
#kubectl get cs
root@argela:/Nebi/psr/kubernetes# kubectl get cs
NAME                 STATUS      MESSAGE                                                                                           ERROR
etcd-4               Unhealthy   Get https://192.168.21.91:2379/health: dial tcp 192.168.21.91:2379: connect: connection refused
etcd-5               Unhealthy   Get https://192.168.21.68:2379/health: dial tcp 192.168.21.68:2379: connect: connection refused
scheduler            Healthy     ok
controller-manager   Healthy     ok
etcd-2               Healthy     {"health":"true"}
etcd-1               Healthy     {"health":"true"}
etcd-0               Healthy     {"health":"true"}
etcd-3               Healthy     {"health":"true"}

ETCD logs while etcd cluster goes to fail : [etcd_docker_log.log](https://github.com/rancher/rke/files/4814098/etcd_docker_log.log)

------------------------------------------------------------------------------

**RKE version:**

INFO[0000] Running RKE version: v1.0.6
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.11", GitCommit:"d94a81c724ea8e1ccc9002d89b7fe81d58f89ede", GitTreeState:"clean", BuildDate:"2020-03-12T21:00:06Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}

**Docker version: (`docker version`,`docker info` preferred)**
Client: Docker Engine - Community
 Version:           19.03.7
 API version:       1.39 (downgraded from 1.40)
 Go version:        go1.12.17
 Git commit:        7141c199a2
 Built:             Wed Mar  4 01:24:10 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.5
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       e8ff056
  Built:            Thu Apr 11 04:13:40 2019
  OS/Arch:          linux/amd64
  Experimental:     false

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
CentOS Linux release 7.8.2003 (Core)
3.10.0-1127.el7.x86_64 
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**

**cluster.yml file:**
nodes:
- address: 192.168.21.110
  port: "22"
  internal_address: 192.168.21.110
  role:
  - controlplane
  - etcd
  - worker
  hostname_override: "master1"
  user: psr
  ssh_key_path: "../argela.pem"
- address: 192.168.20.194
  port: "22"
  internal_address: 192.168.20.194
  role:
  - controlplane
  - etcd
  - worker
  hostname_override: "master2"
  user: psr
  ssh_key_path: "../argela.pem"
- address: 192.168.21.109
  port: "22"
  internal_address: 192.168.21.109
  role:
  - controlplane
  - etcd
  - worker
  hostname_override: "worker1"
  user: psr
  ssh_key_path: "../argela.pem"
- address: 192.168.21.93
  port: "22"
  internal_address: 192.168.21.93
  role:
  - controlplane
  - etcd
  - worker
  hostname_override: "worker2"
  user: psr
  ssh_key_path: "../argela.pem"
- address: 192.168.21.91
  port: "22"
  internal_address: 192.168.21.91
  role:
  - controlplane
  - etcd
  - worker
  hostname_override: "worker3"
  user: psr
  ssh_key_path: "../argela.pem"
- address: 192.168.21.68
  port: "22"
  internal_address: 192.168.21.68
  role:
  - controlplane
  - etcd
  - worker
  hostname_override: "worker4"
  user: psr
  ssh_key_path: "../argela.pem"
services:
  kube-api:
    service_cluster_ip_range: 10.43.0.0/16
    pod_security_policy: false
    always_pull_images: false
  kube-controller:
    cluster_cidr: 10.42.0.0/16
    service_cluster_ip_range: 10.43.0.0/16
    always_pull_images: false
  kubelet:
    cluster_domain: cluster.local
    cluster_dns_server: 10.43.0.10
    fail_swap_on: false
network:
  plugin: canal
authentication:
  strategy: x509
ssh_key_path: "../argela.pem"
ssh_agent_auth: no
authorization:
  mode: rbac
ignore_docker_version: false
kubernetes_version: "v1.15.11-rancher1-2"
private_registries:
private_registries:
- url: 192.168.240.251:10001
  user: docker
  password: docker
  is_default: true
cluster_name: "psr"
restore:
  restore: false
  snapshot_name: ""

**Steps to Reproduce:**

**Results:**

**Comment by superseb** [comment]
etcd is a quorum based system, it needs a majority to survive as described on https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/production/recommended-architecture/#number-of-etcd-nodes. Let me know if that doesn't solve the issue.

**Comment by unlenen** [comment]
@superseb  thanks for answer. We have 20 machines whis are seperated to 2 different regions homogenly . We wish to create a HA kubernetes platform so if we lost one of regions , all pods should continue give services. We made several tests and have seen services are ok but we can not access our cluster in such a scenario until problem is resolved at region . Is it possible for you to recommend us a solution with rke?   We also tried to take a backup and restore to a new master which is described at link (https://rancher.com/docs/rke/latest/en/etcd-snapshots/example-scenarios/#2-simulate-a-node-failure)  but not worked properly
